.github-workflow-script

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+import yaml
+import argparse
+
+# Parse command-line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("workflow_file",
+                    help="Path to the GitHub Actions workflow file")
+
+args = parser.parse_args()
+
+# Load the GitHub Actions workflow file as YAML
+with open(args.workflow_file, "r") as file:
+    data = yaml.safe_load(file)
+
+for step in data["jobs"]["check_python"]["steps"]:
+    if "env" in step and "KCIDB_HOOKS" in step["env"]:
+        print(f"\n# { step['name']}")
+        if "run" in step:
+            command = step["run"]
+            # If the command is a list, join the items into a string
+            if isinstance(command, list):
+                command = "\n".join(command)
+            print(command)
+
+# Extract steps with an `env` tag defined in the `check_auxiliary_files` job
+for step in data["jobs"]["check_auxiliary_files"]["steps"]:
+    if "env" in step and "KCIDB_HOOKS" in step["env"]:
+        print(f"\n# {step['name']}")
+        if "run" in step:
+            command = step["run"]
+            # If the command is a list, join the items into a string
+            if isinstance(command, list):
+                command = "\n".join(command)
+            print(command)
```

Comment:
You don't need to do these separately, and it actually hurts the flexibility by picking only these two specific jobs. Just have two nested loops: the outer going over all the jobs, and the inner going over all the steps in the job, and rely fully on the `KCIDB_HOOKS` env variable to tell you what to execute.